test/controllers/CharityNameDetailsControllerSpec.scala

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import helpers.TestHelper
+import play.api.http.Status
+
+
+class CharityNameDetailsControllerSpec extends TestHelper  {
+
+  lazy val charityNameDetailsController = fakeApplication.injector.instanceOf[CharityNameDetailsController]
```

Comment:
I would prefer if we didn't use a fake application to inject an instance of the controller but created one with mocked variables. This applies for the other tests. This controller also needs tests for auth.